subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformedFileCache.java

```diff
@@ -161,6 +180,29 @@ private CacheKey getCacheKey(File inputFile, HashCode inputsHash) {
         return new CacheKey(inputFileSnapshot, inputsHash);
     }
 
+    @SuppressWarnings("Since15")
+    private void markAccessedByTouchingTopmostOutputDirectory(List<File> files) {
+        if (files.isEmpty()) {
+            return;
+        }
+        Path baseDir = filesOutputDirectory.toPath();
+        // all files share the same output directory so only taking into account the first one
+        Path relativePath = baseDir.relativize(files.get(0).toPath());
+        if (relativePath.getNameCount() >= 2) {
+            // touch output dir of accessed files so cache cleanup does not evict them
+            touch(baseDir.resolve(relativePath.subpath(0, 2)));
+        }
+    }
+
+    @SuppressWarnings("Since15")
+    private static void touch(Path file) {
+        try {
+            Files.setLastModifiedTime(file, FileTime.fromMillis(System.currentTimeMillis()));
```

Comment:
⭕️ Guava has `Files.touch`.

Reply:
Since I already have a `Path` I didn't want to convert it to a `File`. If that's alright with you, I'll leave it as is.